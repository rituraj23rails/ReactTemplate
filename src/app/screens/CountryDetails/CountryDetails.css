.countryDetailsMainContainer {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.countryDetail {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-top: 20px;
}

.countryDetailText {
    font-size: 25px;
    color: #000;
    font-weight: bold;
}

.countryDetailCell {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 2px solid #000;
    width: 600px;
    padding: 10px 30px;
    margin: 20px auto 20px auto;
}

.dataValue {
    font-size: 15px;
    font-weight: bold;
    margin-bottom: 10px;
}

.flagIcon {
    width: 200px;
    height: 150px;
    margin-bottom: 10px;
}

.weatherButton {
    width: 200px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid #000;
    margin-top: 50px;
    font-size: 15px;
    font-weight: bold;
}

.weatherButton:hover {
    background-color: green;
    color: #fff;
}

.weatherContainer {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin-top: 20px;
    border: 2px solid #000;
    padding: 20px;
    margin-bottom: 20px;
}

.weatherIcon {
    width: 100px;
    height: 100px;
    margin-left: 20px;
}

@media screen and (max-width: 820px) {
    
    .countryDetailCell {
        width: 400px;
    }
    
}

@media screen and (max-width: 480px) {
    
    .countryDetailCell {
        width: 300px;
    }
    
}
