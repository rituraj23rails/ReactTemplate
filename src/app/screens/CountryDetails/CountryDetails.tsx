import React, { Component, useRef } from 'react';
import './CountryDetails.css';
import WithRouter from '../../../WithRouter';

export interface Props {
    navigate: any;
    location: any;
    id: string;
}

interface S {
    weatherData: any;
    showData: any;
}

interface SS {
    id: any;
}

class CountryDetails extends Component<Props, S, SS> {
    carousel: any = null;
    constructor(props: any) {
        super(props);
        this.state = {
            weatherData: null,
            showData: false,
        };
    };

    componentDidMount = async () => {
        console.log('@@@ Params =========', this.props.location.state.countryData)
    }

    componentWillUnmount = () => {

    }

    onCalculateCapitalWeather = async () => {
        const { capital } = this.props.location.state.countryData;
        let cap = capital[0];
        try {
            let response = await fetch(
                `http://api.weatherstack.com/current?access_key=e32b270172beca34f23673646d8a9b0d&query=${cap}`,
            );

            let json = await response.json();
            this.setState({ weatherData: json.current }, () => {
                this.setState({ showData: true });
                console.log('@@@ Weather DATA JSON ============', this.state.weatherData);
            })
        } catch (error) {
            console.log('@@@ Error Weather Data =========', error);
        }
    }

    renderCountryDetails = () => {
        const { name, capital, population, latlng = ['0', '0'], flags } = this.props.location.state.countryData;
        return (
            <div className="countryDetail">
                <p className="countryDetailText">Country Details</p>
                <div className="countryDetailCell">
                    <img src={flags?.svg} className="flagIcon" />
                    <div>
                        <p className="dataValue">Country: {name?.common}</p>
                        {capital && <p className="dataValue">Capital: {capital[0] ? capital[0] : ''}</p>}
                        <p className="dataValue">Population: {population}</p>
                        <p className="dataValue">Latlng: {latlng[0]} {latlng[1]}</p>
                    </div>
                </div>
                <button onClick={() => this.onCalculateCapitalWeather()} className="weatherButton">Capital Weather</button>
            </div>
        )
    }

    renderWeatherData = () => {
        const { wind_speed, temperature, precip, weather_icons, pressure } = this.state.weatherData;
        return (
            <div className="weatherContainer">
                <div>
                    <p className="dataValue">Temperature: {temperature}</p>
                    <p className="dataValue">PRECIP: {precip}</p>
                    <p className="dataValue">Wind Speed: {wind_speed}</p>
                </div>
                <img src={weather_icons[0]} className="weatherIcon" />
            </div>
        )
    }

    render() {
        return (
            <div className='countryDetailsMainContainer'>
                {this.renderCountryDetails()}
                {this.state.showData && this.renderWeatherData()}
            </div>
        )
    }
};

export default WithRouter(CountryDetails);
export {CountryDetails};