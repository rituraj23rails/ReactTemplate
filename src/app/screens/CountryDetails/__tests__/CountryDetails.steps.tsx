import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CountryDetails } from "../CountryDetails";

configure({adapter: new Adapter()});

const screenProps = {
    navigate: jest.fn(),
    location: {
        state: {
            countryData: []
        }
    },
    id: "CountryDetails"
}

const feature = loadFeature('./src/app/screens/CountryDetails/__tests__/features/CountryDetails-scenarios.feature');

defineFeature(feature, (test) => {

    test('User navigates Country Details Screen', ({ given, when, then }) => {
        let CountryDetailsScreenWrapper: ShallowWrapper;
        let CountryDetailsScreenInstance: CountryDetails;

        given('I am a User loading Country Details Screen', () => {
            CountryDetailsScreenWrapper = shallow(<CountryDetails {...screenProps} />)
        });

        when('I navigate to the Country Details Screen', () => {
            CountryDetailsScreenInstance = CountryDetailsScreenWrapper.instance() as CountryDetails
        });

        then('Country Details Screen will load with out errors', () => {
            CountryDetailsScreenInstance.componentDidMount();
            expect(CountryDetailsScreenWrapper).toBeTruthy()
            expect(CountryDetailsScreenWrapper).toMatchSnapshot()
        });

        then('I can leave the screen with out errors', () => {
            CountryDetailsScreenInstance.componentWillUnmount();
            expect(CountryDetailsScreenWrapper).toBeTruthy()
            expect(CountryDetailsScreenWrapper).toMatchSnapshot()
        });
    });

});
