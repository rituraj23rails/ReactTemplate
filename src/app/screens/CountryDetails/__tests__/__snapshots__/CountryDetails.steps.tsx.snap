// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Details Screen User navigates Country Details Screen 1`] = `
<div
  className="countryDetailsMainContainer"
>
  <div
    className="countryDetail"
  >
    <p
      className="countryDetailText"
    >
      Country Details
    </p>
    <div
      className="countryDetailCell"
    >
      <img
        className="flagIcon"
      />
      <div>
        <p
          className="dataValue"
        >
          Country: 
        </p>
        <p
          className="dataValue"
        >
          Population: 
        </p>
        <p
          className="dataValue"
        >
          Latlng: 
          0
           
          0
        </p>
      </div>
    </div>
    <button
      className="weatherButton"
      onClick={[Function]}
    >
      Capital Weather
    </button>
  </div>
</div>
`;

exports[`Country Details Screen User navigates Country Details Screen 2`] = `
<div
  className="countryDetailsMainContainer"
>
  <div
    className="countryDetail"
  >
    <p
      className="countryDetailText"
    >
      Country Details
    </p>
    <div
      className="countryDetailCell"
    >
      <img
        className="flagIcon"
      />
      <div>
        <p
          className="dataValue"
        >
          Country: 
        </p>
        <p
          className="dataValue"
        >
          Population: 
        </p>
        <p
          className="dataValue"
        >
          Latlng: 
          0
           
          0
        </p>
      </div>
    </div>
    <button
      className="weatherButton"
      onClick={[Function]}
    >
      Capital Weather
    </button>
  </div>
</div>
`;
